packages/airnode-protocol-v1/contracts/dapis/DapiServer.sol

```diff
@@ -451,6 +454,32 @@ contract DapiServer is
         emit UpdatedDapiWithBeacons(dapiId, updatedValue, updatedTimestamp);
     }
 
+    /// @notice Updates the dAPI that is specified by the beacon IDs and
+    /// returns if this update was justified according to the update threshold
+    /// @dev This method does not allow the caller to indirectly read a dAPI,
+    /// which is why it does not require the sender to be a void signer with
+    /// zero address. This allows the implementation of incentive mechanisms
+    /// that rewards keepers that trigger valid dAPI updates.
+    /// @param beaconIds Beacon IDs
+    /// @param updateThresholdInPercentage Update threshold in percentage where
+    /// 100% is represented as `HUNDRED_PERCENT`
+    function updateDapiWithBeaconsAndReturnCondition(
+        bytes32[] memory beaconIds,
+        uint256 updateThresholdInPercentage
+    ) public override returns (bool) {
+        bytes32 dapiId = keccak256(abi.encode(beaconIds));
+        DataPoint memory initialDataPoint = dataPoints[dapiId];
+        updateDapiWithBeacons(beaconIds);
+        DataPoint storage updatedDataPoint = dataPoints[dapiId];
+        return
+            calculateUpdateInPercentage(
+                initialDataPoint.value,
+                dataPoints[dapiId].value
+            ) >=
+            updateThresholdInPercentage ||
```

Comment:
is there any difference between using `updateThresholdInPercentage` vs just using `conditionParameters` like in `conditionPspBeaconUpdate` ? You can derive the updateThreshold from the condition parameters.

Reply:
ok that's the only comment I had, everything else looks good 👍